### PR TITLE
Mean Absolute Percentage Error Implementation

### PR DESCRIPTION
**Pull request by Ciroye**
Add the mean absolute percentage error implementation in base of mean absolute error module and the current implementation in tensorflow-keras library

### CODE REVIEW

**Comment by Ciroye** [comment]
Sure, I'm going to work in that tomorrow and I'll make the upload of the test file

File: elegy/losses/mean_absolute_percentage_error.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    
+    Arguments:
+        y_true: Ground truth values. shape = `[batch_size, d0, .. dN]`.
+        y_pred: The predicted values. shape = `[batch_size, d0, .. dN]`.
+
+    Returns:
+        Mean absolute percentage error values. shape = `[batch_size, d0, .. dN-1]`.
+    """
+    y_true = y_true.astype(y_pred.dtype)
+    diff = jnp.abs((y_pred - y_true) / jnp.clip(y_true, jnp.finfo(float).eps, None))
```
Comment (cgarciae) [inline elegy/losses/mean_absolute_percentage_error.py:38]: Change `jnp.finfo(float).eps` to `utils.EPSILON`

**Comment by cgarciae** [comment]
@Ciroye Nice work! As @charlielito said we are missing tests for this metric, you can use one of the other test files as a template.

**Comment by Ciroye** [comment]
I've already create the test file and proved it in my own environment, if you have any doubt or comment just let me know

File: elegy/losses/mean_absolute_percentage_error_test.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    assert mape(y_true, y_pred) == 2.7777786
+
+    # Calling with 'sample_weight'.
+    assert mape(y_true, y_pred, sample_weight=jnp.array([0.1, 0.9])) == 2.5000007
+
+    # Using 'sum' reduction type.
+    mape = elegy.losses.MeanAbsolutePercentageError(reduction=elegy.losses.Reduction.SUM)
+
+    assert mape(y_true, y_pred) == 5.5555573
```
Comment (cgarciae) [inline elegy/losses/mean_absolute_percentage_error_test.py:33]: Change `==` for `jnp.isclose` with some `rtol` value on all asserts.

File: elegy/losses/mean_absolute_percentage_error_test.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    y_true = jax.random.randint(rng, shape=(2, 3), minval=0, maxval=2)
+    y_pred = jax.random.uniform(rng, shape=(2, 3))
+    y_true = y_true.astype(y_pred.dtype)
+
+    loss = elegy.losses.mean_percentage_absolute_error(y_true, y_pred)
+
+    assert loss.shape == (2,)
+
+    assert jnp.array_equal(loss, 100 * jnp.mean(jnp.abs((y_pred - y_true) / (jnp.clip(y_true, utils.EPSILON, None))), axis=-1))
```
Comment (cgarciae) [inline elegy/losses/mean_absolute_percentage_error_test.py:53]: Can you test against the `tf.keras` implementation as done here?

https://github.com/poets-ai/elegy/blob/9fc946d152f6064f1d2d71a726754d0df31b09d7/elegy/losses/binary_crossentropy_test.py#L36

Comment (Ciroye) [reply elegy/losses/mean_absolute_percentage_error_test.py:63]: I'd already add the keras verification and used the isclose method

File: elegy/losses/mean_absolute_percentage_error.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            reduction: (Optional) Type of `elegy.losses.Reduction` to apply to
+                loss. Default value is `SUM_OVER_BATCH_SIZE`. For almost all cases
+                this defaults to `SUM_OVER_BATCH_SIZE`. When used with
+                `tf.distribute.Strategy`, outside of built-in training loops such as
+                `elegy` `compile` and `fit`, or `SUM_OVER_BATCH_SIZE`
+                will raise an error.
+                for more details.
+            name: Optional name for the loss.
+            weight: Optional weight contribution for the total loss. Defaults to `1`.
+        """
```
Comment (cgarciae) [inline elegy/losses/mean_absolute_percentage_error.py:104]: Please add documentation for the `on` argument, use the same as in other metrics.
